spec/requests/groups_controller_spec.rb

```diff
@@ -858,6 +858,54 @@ def expect_type_to_return_right_groups(type, expected_group_ids)
 
       context "user default notifications" do
         it "should update default notification preference for existing users" do
+          user1 = Fabricate(:user)
+          user2 = Fabricate(:user)
+          group.add(user1)
+          group.add(user2)
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            }
+          }
+
+          expect(response.status).to eq(200)
+          group_users = group.group_users
+          expect(response.parsed_body["user_count"]).to eq(group_users.count)
+
+          group_users.where(user: user1).update_all(notification_level: 1)
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            }
+          }
+
+          expect(response.status).to eq(200)
+          expect(response.parsed_body["user_count"]).to eq(group.group_users.count - 1)
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            },
+            update_existing_users: true
+          }
+
+          expect(response.status).to eq(200)
+          expect(response.parsed_body["success"]).to eq("OK")
+          expect(GroupUser.exists?(user: user2, group: group, notification_level: 2)).to be_truthy
```

Comment:
Should this assertion be against `user1` instead of `user2` since we changed `user1`'s notification level in a prior request?

Reply:
No, it's correct. `user1` is not affected by this update since it has a custom notification level now.

Reply:
But `user2`'s notification level has already been updated to `2` in a prior request, why will this request update `user2`'s notification level again?

Reply:
Nope. That's where you misunderstood. In the previous request, it will return only the user count (number of users who are going to be updated). It will update existing users only if the `update_existing_users: true` param is available.

Anyway, I added one more important fix to correctly skip existing users.